### PR TITLE
Map PV access modes to CSI access modes

### PR DESCRIPTION
**Pull request by chrishenzie**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR adds functions for mapping PersistentVolume access modes in Kubernetes to CSI access modes.

Similar logic was added in the external-provisioner, external-attacher, and external-resizer. I'm moving this logic into this shared library to reduce redundancy. As a followup I will need to update the PRs for those sidecars to use an updated release of this library.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
The K8s API change and feature addition [PR#102028 is here](https://github.com/kubernetes/kubernetes/pull/102028) and still is in the early stages of review. Until then, I have a placeholder commit using my forked version of the K8s API.

**Does this PR introduce a user-facing change?**:
```release-note
Adds mappings for PV access modes to new CSI access modes: `SINGLE_NODE_SINGLE_WRITER` and `SINGLE_NODE_MULTI_WRITER`.
```

/assign @msau42 
/assign @jingxu97

### CODE REVIEW

**Comment by chrishenzie** [comment]
@jsafrane @msau42 

Follow-up to enforcing `ReadWriteOncePod` can only be used in isolation. I made a similar change here in the last commit. Let me know if this works and I will squash it into the parent commit.

**Comment by jsafrane** [comment]
Looks good, if "`ReadWriteOncePod` can only be used in isolation" is approved in API review.

**Comment by chrishenzie** [comment]
Bumped k8s to v1.22.0-beta.1. Waiting on release of v1.22.

/hold

**Comment by chrishenzie** [comment]
/hold cancel

Discussed this in CSI standup (7/14), we're going to merge this with the beta version dependency and bump it to the release once it's cut, PTAL

File: go.mod [hunk 1]
```diff
@@ -2,24 +2,25 @@ module github.com/kubernetes-csi/csi-lib-utils
...
-	google.golang.org/grpc v1.37.0
-	k8s.io/api v0.21.1
-	k8s.io/client-go v0.21.1
+	github.com/golang/protobuf v1.5.2
+	github.com/stretchr/testify v1.7.0
+	golang.org/x/net v0.0.0-20210520170846-37e1c6afe023
+	google.golang.org/grpc v1.38.0
+	k8s.io/api v0.22.0-beta.1
+	k8s.io/client-go v0.22.0-beta.1
 	k8s.io/component-base v0.21.1
```
Comment (msau42) [inline go.mod:14]: should this also be updated?

**Review by msau42 — COMMENTED** [review]
/approve

Will let @jsafrane review

**Comment by chrishenzie** [comment]
Bumping, @jsafrane when you have a moment to review

**Pull request merged**